Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -313,20 +313,34 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
       return information;
     }
 
-    private string FormatResourceCount(int nodeResourceCount) {
+    public static string FormatResourceCount(long nodeResourceCount) {
       var suffix = 0;
+      var fractional = 0;
+      // First, round up nodeResourceCount to 3 significant digits.
```

Comment:
Could you introduce a `roundToDigits` function so we don't need the comment? Also, would another way of rounding be to divide by `nDigits - 3`, then to do `Math.floor(x + 0.5)`, and then multiply by `nDigits - 3` ?

Reply:
> Could you introduce a roundToDigits function so we don't need the comment? 

Done

> Also, would another way of rounding be to divide by nDigits - 3, then to do Math.floor(x + 0.5), and then multiply by nDigits - 3 ?

![image](https://github.com/dafny-lang/dafny/assets/3601079/91a1dbcb-6196-494e-a040-3145658c1cad)

Indeed. I'll keep my version because it contains only one modulo compared to two division/multiplication in base 10 :-)